src/components/FileDropzone/MucFileDropzone.vue

```diff
@@ -0,0 +1,278 @@
+]);
+
+/** Flag signals if file size is valid */
+const validFileSize = ref(true);
+
+/** Flag signals if files where dragged over the drop zone */
+const isDragOver = ref(false);
+
+/** Hidden HTML input field that opens the file explorer */
+const fileInput = document.createElement("input");
+
+/**
+ * Configuration of the HTML input field.
+ */
+onMounted(() => {
+  fileInput.type = "file";
+  fileInput.multiple = true;
+  /* catch 'onchange' event and trigger emit to the surrounding element */
+  fileInput.onchange = (event) => {
+    const target = event.target as HTMLInputElement;
+    if (target?.files && target.files.length > 0) {
+      const filesArray = Array.from(target.files);
+      _emitFiles(filesArray);
+    }
+  };
+});
+
+/**
+ * Opens the file explorer by firing a click event to the hidden file input field if the property {@link Props#disabled} is false.
+ */
+const selectFiles = () => {
+  if (props.disabled) return;
+  fileInput.click();
+};
+
+/** Sets flag {@link isDragOver} true. */
+const onDragOver = () => (isDragOver.value = true);
+
+/** Sets flag {@link isDragOver} false. */
+const onDragLeave = () => (isDragOver.value = false);
+
+/**
+ * Catches a DragEvent in the drop zone and emits dropped files to the surrounding element if the property {@link Props#disabled} is false.
+ * @param {DragEvent} event dropped files
+ */
+const onDrop = (event: DragEvent) => {
+  if (props.disabled) return;
+  isDragOver.value = false;
+  const dataTransfer: DataTransfer = event.dataTransfer as DataTransfer;
+  if (dataTransfer?.files?.length > 0) {
+    const filesArray = Array.from(dataTransfer.files);
+    _emitFiles(filesArray);
+  }
+};
+
+/**
+ * Checks if all files are inside the allowed file size range that is given by {@link Props#maxFileSize}.
+ * Reads the file contents and converts it to  Base64-encoded data preceded by a Data-URL declaration.
+ * Emits the converted files to upload to the surrounding element.
+ * @param {File[]} files array of dropped or chosen files to upload
+ */
+const _emitFiles = (files: File[]) => {
+  if (!_areFilesValid(files)) {
+    validFileSize.value = false;
+    return;
+  }
+  validFileSize.value = true;
+
+  const fileDataPromises = files.map((file) => {
+    return new Promise<FileDTO>((resolve) => {
+      const reader = new FileReader();
+      reader.onload = () => {
+        const fileDTO = new FileDTO();
+        fileDTO.filename = file.name;
+        fileDTO.mimeType = file.type;
+        fileDTO.data = reader.result as string;
+        resolve(fileDTO);
+      };
+      reader.readAsDataURL(file);
+    });
+  });
+  Promise.all(fileDataPromises).then((files) => {
+    emit("files", files);
+  });
+};
+
+/**
+ * Checks if all files are inside the allowed file size range that is given by {@link Props#maxFileSize}.
+ * @param {File[]} files array files
+ */
+const _areFilesValid = (files: File[]) => {
+  return !files.some((file) => file.size > props.maxFileSize * 1024 * 1024);
```

Comment:
Debatable: Is the maximum filesize checked per file or on all uploaded files combined?

Reply:
At the moment, each file is being checked for size individually. According to the current requirements, each file, and not the total amount, should not exceed a maximum size.

What do you think would be sensible?

Reply:
Hard to decide. Thinking about our current use-case: the api of the receiving system doesn't allow uploading files seperately - only in one big json object. In this case it would be beneficial to limit the size of all files together.

If you have a clean api with which you can upload each file sperately this wouldn't be helpful.

Maybe removing the validation part of this component completly would be the best way? The surrounding component can than use the files from the event to validate and the `disable`-Flag of the upload-component to control the validation.

Reply:
I removed file size validation completely. This is the job of the surrounding element from now on.